### DISCUSSION TITLE
Debugger can't access local variables?

### DISCUSSION THREAD

**Discussion by ppazos** [nextgenhealthcare/connect#6089] [Feature/Enhancement Ideas]
I'm running the debugger and set a breakpoint on a transformer script.

Then tried to access the local variable `msg` in the expression panel, and when the execution stops at my breakpoint, the expression value says `"msg" is not defined`.

How can I watch local variables (mine or Mirth's e.g. msg, tmp) and display its internal structure?

Thanks!

![debugger](https://github.com/nextgenhealthcare/connect/assets/1278556/44761cd8-c6ca-48a9-a6c0-057bcee8b871)

**Comment by pacmano1** [comment]
Seems to work for me:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/99970b42-7513-44aa-b93f-e17d947389fb)

**Reply by pacmano1** [reply]
Also tried that with a mapper step and it worked fine.  Looks like you were using a mapper step vs JS step.

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/7a612e44-0eae-4773-8542-f9d4ea726a0c)

**Reply by ppazos** [reply]
Thanks @pacmano1 can you mention the steps you followed?

I did this:

1. set the breakpoint
2. sent a message to the channel
3. when the execution stopped at the breakpoint, I added the names for the watched expressions and the values where "is not defined"

Should I set the expressions before sending the message?

**Reply by pacmano1** [reply]
I did what you did.
